Packages/io.chainsafe.web3-unity/Runtime/Scripts/Connection/WalletConnectConnectionProvider.cs

```diff
@@ -11,36 +17,122 @@ namespace ChainSafe.Gaming.UnityPackage.Connection
     /// WalletConnect connection provider used for connecting to a wallet using WalletConnect.
     /// </summary>
     [CreateAssetMenu(menuName = "ChainSafe/Connection Provider/Wallet Connect", fileName = nameof(WalletConnectConnectionProvider))]
-    public class WalletConnectConnectionProvider : RestorableConnectionProvider
+    public class WalletConnectConnectionProvider : RestorableConnectionProvider, IWalletConnectConfig, IConnectionHandlerProvider
```

Comment:
Why is hyperplay connection provider not also the IConnectionHandlerProvider?

Reply:
`IConnectionHandlerProvider` is specific to wallet connect, naming is a bit ambiguous

Reply:
Yeah maybe change the name to `IWallectConnetConnectionHandler?`